transport/roundtrip_test.go

```diff
@@ -249,3 +278,81 @@ func TestSimpleRoundTrip(t *testing.T) {
 		}
 	}
 }
+
+func TestSimpleRoundTripOneway(t *testing.T) {
+	trans := httpTransport{t}
+
+	tests := []struct {
+		name           string
+		requestHeaders transport.Headers
+		requestBody    string
+	}{
+		{
+			name:           "hello world",
+			requestHeaders: transport.NewHeaders().With("foo", "bar"),
+			requestBody:    "hello world",
+		},
+		{
+			name:           "empty",
+			requestHeaders: transport.NewHeaders(),
+			requestBody:    "",
+		},
+	}
+
+	rootCtx := context.Background()
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			requestMatcher := transporttest.NewRequestMatcher(t, &transport.Request{
+				Caller:    testCaller,
+				Service:   testService,
+				Procedure: testProcedureOneway,
+				Encoding:  raw.Encoding,
+				Headers:   tt.requestHeaders,
+				Body:      bytes.NewReader([]byte(tt.requestBody)),
+			})
+
+			handlerDone := make(chan struct{})
+
+			onewayHandler := onewayHandlerFunc(func(_ context.Context, r *transport.Request) error {
+				assert.True(t, requestMatcher.Matches(r), "request mismatch: received %v", r)
+
+				// Pretend to work: this delay should not slow down tests since it is a
+				// server-side operation
+				time.Sleep(5 * time.Second)
+
+				// fill the channel, telling the client (which should not be waiting for
+				// a response) that the handler finished executing
+				handlerDone <- struct{}{}
```

Comment:
I would close the channel rather than filling it. This frees up the resource
plus it wakes up ALL waiting goroutines rather than just one.